tools/clone_flutter.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -e
+
+
+# Go to the engine git repo to get the date of the latest commit.
+cd $ENGINE_PATH/src/flutter
+# Get latest commit's time for the engine repo.
+# Use date based on local time otherwise timezones might get mixed.
+LATEST_COMMIT_TIME_ENGINE=`git log -1 --date=local --format="%ad"`
+echo "Latest commit time on engine found as $LATEST_COMMIT_TIME_ENGINE"
+
+# Do rest of the task in the root directory
+cd ~
+mkdir -p $FRAMEWORK_PATH
+cd $FRAMEWORK_PATH
+# Clone the Flutter Framework.
+git clone https://github.com/flutter/flutter.git
+cd flutter
+# Get the time of the youngest commit older than engine commit.
+# Before makes the comparison considering the timezone as well.
+COMMIT_NO=`git log --before="$LATEST_COMMIT_TIME_ENGINE" -n 1 | grep commit | cut -d ' ' -f2`
```

Comment:
What happens if the latest engine commit is after the latest framework commit?  Will this say there's no commit to reset to?  If it leaves the framework at ToT, then multiple runs of this script at the same engine commit could still yield different framework versions...

Reply:
This is deterministically getting the youngest framework commit older than the engine commit. For example engine commit is 10pm PST. The framework tree was red for half a day. The framework commit will be 10am PST.

Yes you are right. Different commits running at the same time on cirrus, since their head is different, will be fetching a different version of the framework repo.

Reply:
Ah cool - at first I read it as "oldest commit younger than the engine commit".

Reply:
So yes, next question is about branches.  How does this behave if you're running on a release branch in the engine?  We may want to have this detect if you're not on master, and then fetch the branch by the same name in the framework (and fail if there's no branch by the same name in the framework).

Reply:
Should we worry about https://stackoverflow.com/questions/12934044/git-parent-commit-is-younger-than-descendant? Perhaps topological age should take precedence over actual date?

@tvolkert How do we run tests on CI for branches? Also, how do we keep LUCI recipes stable? A change in LUCI recipe can break our tests in older commits too (in fact it did just last week).

Reply:
On flutter/engine there isn't any branches for 'dev', 'beta', 'stable'. Is the suggestion for managing the hotfixes?

Reply:
@yjbanov On stackoverflow article:

1) The commit date can be changed that is true. On the other hand we are simply trying to prevent both trees being closed on an error on flutter/flutter. We can add this to the Guidelines and simply ask the engine contributors not to change the date. 
2) git log --before and --after comments are always using commit date, so all the other parts on the stackoverflow discussion regarding the rebases are irrelevant to line 21 of the script.

Reply:
@yjbanov infra team is working on the ability to pin the recipe version to the engine.

@nturgut these won't be dev, beta, or stable - they'll be release branches (one per dev release, created on both the framework and the engine).

Reply:
@Hixie  @tvolkert I added the checks for the branch. @yjbanov I added a comment for your concern.

Even though this script works locally, I am not sure how it will run for the release process with cirrus CI. as far I as understand cirrus only runs on detached head. 'master' branches is used for post submit benchmarks. Once the cloning is done, there isn't any other configuration for a specific branch.

In my opinion we should either use LUCI for the release process or change the clone script of cirrus https://cirrus-ci.org/guide/tips-and-tricks/#custom-clone-command

Reply:
Looking at the github history I couldn't find any instances when a child commit date is older than the parent's, so it looks like Github's "squash and merge" button does the right thing for us. This can only happen if someone pushes using the command line, and they would really have to go out of their way to change the dates. I think we're good.

Recipe pinning SGTM. Or better, move them into the respective repos.